subxt/src/utils/mod.rs

```diff
@@ -26,7 +28,7 @@ pub use primitive_types::{H160, H256, H512};
 
 /// Wraps an already encoded byte vector, prevents being encoded as a raw byte vector as part of
 /// the transaction payload
-#[derive(Clone, Debug, Eq, PartialEq)]
+#[derive(Clone, Debug, Eq, PartialEq, Decode)]
```

Comment:
Ah hmmm, I'm wondering about this automatic `Decode`, because:

- decoding like this will first decode a Compact length for the vec length and then put the relevant number of butes into the vec.
- encoding is not the opposite of this; it will just encode the bytes in the vec to the target; no compact length or whatever at the beginning.

Having a look, an extrinsic is actually just a compact encoded length and then that number of bytes.

So probably you were right in a previous iteration, and `UncheckedExtrinsic` should have a custom `Decode` impl that will decode the compact length and then the relevant number of bytes following it all into the `Static<Encoded>` struct, which will then encode (from the impl below) precisely those bytes back.